safehttp/response_writer_test.go

```diff
@@ -50,3 +53,57 @@ func TestResponseWriterSetInvalidCookie(t *testing.T) {
 		t.Error("rw.SetCookie(c) got: nil want: error")
 	}
 }
+
+func TestResponseWriterWriteTwicePanic(t *testing.T) {
+	tests := []struct {
+		name  string
+		write func(w *safehttp.ResponseWriter)
+	}{
+		{
+			name: "Call Write twice",
+			write: func(w *safehttp.ResponseWriter) {
+				w.Write(safehtml.HTMLEscaped("<h1>Escaped, so not really a heading</h1>"))
```

Comment:
Suggestion: make `write` be the full "handler", i.e. in this example, do `w.Write()` twice.

Why? This way you can also add tests for "400, then 200" cases (i.e. `w.ServerError(); w.Write()`). Also, it feels somewhat surprising to see the description say "Call Write twice" and then a line below see it called once.

Reply:
I agree it is a little confusing, but I want to be able to defer the recover between the writes, so that it only detects if the seconds write panics. If I make the whole "handler" part of the table then the only part left in the loop is creating the `ServeMux`. Like this:
-Handler: Write, Defer, Write.
-Loop: Create ServeMux, call handler.
In that case it would make more sense to just create a separate test for each case and ditch the table test since the tests won't share any logic anymore.

I could have a `write1` and a `write2` in the table.

And should I add all the 5*5 combinations of ways to write twice?

Reply:
It's fine if you do the recover at the beginning. We have other tests to check that things don't panic on a first write.